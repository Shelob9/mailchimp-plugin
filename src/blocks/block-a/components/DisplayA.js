import React from 'react';

export const DisplayA = ({message,useBold = false}) => {
	if( useBold ){
		return <span><strong>{message}</strong></span>
	}

	return <span>{message}</span>
};

