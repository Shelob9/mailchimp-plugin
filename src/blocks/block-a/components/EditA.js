import React from 'react';
import {TextControl,RadioControl} from '@wordpress/components';
export const EditA = ({message,onChangeMessage,useBold,onChangeBold}) =>
	 (
		<div>
			<TextControl
				className={'edit-a-message'}
				label={'The Message'}
				value={message}
				onChange={onChangeMessage}
			/>
			<RadioControl
				className={'edit-a-use-bold'}
				options={ [
					{ label: 'Bold', value: true },
					{ label: 'No Special Style', value: false },
				] }
				label={'Use Bold'}
				value={useBold}
				onChange={(useBold) => {
					onChangeBold((useBold == 'true'))
				}}
			/>
		</div>
	);


