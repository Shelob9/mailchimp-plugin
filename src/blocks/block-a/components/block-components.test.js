import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {DisplayA} from "./DisplayA";
import {EditA} from "./EditA";

describe('Tests working', () => {
	it('should work', () => {
		expect(1).toBe(1);
	});
});

describe('DisplayA component', () => {
	it( 'shows message', () => {
		const component = renderer.create(
			<DisplayA message={'The Words'} />
		);
		expect( component.toJSON()).toMatchSnapshot();
	});

	it( 'shows message in bold', () => {
		const component = renderer.create(
			<DisplayA message={'The Words'} useBold={true} />
		);
		expect( component.toJSON()).toMatchSnapshot();
	})
});


describe('EditA component', () => {
	let onChangeMessage;
	let onChangeBold;
	beforeEach(() => {
		onChangeMessage = jest.fn();
		onChangeBold = jest.fn();
	});
	it( 'Updates message', () => {
		const component = mount(
			<EditA
				onChangeMessage={onChangeMessage}
				onChangeBold={onChangeBold}
				useBold={true}
				message={'Hello'}
			/>
		);
		component.find( '.edit-a-message' ).find( 'input' ).simulate( 'change' );
		expect( onChangeMessage.mock.calls.length ).toBe(1);
		expect( onChangeBold.mock.calls.length ).toBe(0);
	});

	it( 'Updates use bold setting', () => {
		const component = mount(
			<EditA
				onChangeMessage={onChangeMessage}
				onChangeBold={onChangeBold}
				useBold={false}
				message={'Hello'}
			/>
		);
		component.find( '.edit-a-use-bold' ).find( 'input' ).first().simulate('change', { target: { checked: true } });
		expect( onChangeBold.mock.calls.length ).toBe(1);
		expect( onChangeMessage.mock.calls.length ).toBe(0);
	});

});
