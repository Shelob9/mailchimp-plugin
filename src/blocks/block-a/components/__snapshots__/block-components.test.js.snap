// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayA component shows message 1`] = `
<span>
  The Words
</span>
`;

exports[`DisplayA component shows message in bold 1`] = `
<span>
  <strong>
    The Words
  </strong>
</span>
`;
